### PR TITLE
Remove RTCIdentityEvent and RTCIdentityErrorEvent

### PR DESCRIPTION
**Pull request by foolip**
These were only ever implemented in Firefox and removed in 2015:
https://bugzilla.mozilla.org/show_bug.cgi?id=975144
https://github.com/mozilla/gecko-dev/commit/e92b44db9851afda6e3526a4ba5b213a8edaba6c

The events identityresult, peeridentity, idpassertionerror and
idpvalidationerror were also removed then.

The data claiming support in Edge 18 for some of this is wrong,
confirmed by testing Edge 18 with these tests:
https://mdn-bcd-collector.appspot.com/tests/api/RTCPeerConnection/onidentityresult
https://mdn-bcd-collector.appspot.com/tests/api/RTCPeerConnection/onidpassertionerror
https://mdn-bcd-collector.appspot.com/tests/api/RTCPeerConnection/onidpvalidationerror
https://mdn-bcd-collector.appspot.com/tests/api/RTCPeerConnection/onpeeridentity

The data was added in https://github.com/mdn/browser-compat-data/pull/987.

### CODE REVIEW

File: api/RTCPeerConnection.json [hunk 1]
```diff
@@ -2778,195 +2628,6 @@
...
-            "firefox": {
-              "version_added": "22"
-            },
-            "firefox_android": {
-              "version_added": "44"
-            },
-            "ie": {
-              "version_added": false
-            },
-            "opera": [
```
Comment (foolip) [inline api/RTCPeerConnection.json:2804]: This data doesn't make any sense, but to be sure I've disconfirmed it in Opera 77 using https://mdn-bcd-collector.appspot.com/tests/api/RTCPeerConnection/onidentityresult.

File: api/RTCPeerConnection.json [hunk 2]
```diff
@@ -2778,195 +2628,6 @@
...
-              }
-            ],
-            "safari": {
-              "version_added": false
-            },
-            "safari_ios": {
-              "version_added": false
-            },
-            "samsunginternet_android": {
-              "version_added": "6.0"
```
Comment (foolip) [inline api/RTCPeerConnection.json:2831]: This is as implausible as it gets, also disconfirmed with https://mdn-bcd-collector.appspot.com/tests/api/RTCPeerConnection/onidentityresult.

**Comment by foolip** [comment]
This got bigger than I thought, but this + https://github.com/mdn/content/pull/6109 should cover the same cleanup now.

**Review by vinyldarkscratch — APPROVED** [review]
Thank you for digging into these!  I'm definitely in favor of their removal, and now that the MDN docs have been removed, time to remove the BCD entries!

**Pull request merged**